hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -219,15 +219,15 @@ sealed abstract class Block extends Product:
           val newBody = d.body.flattened
           if newBody is d.body
           then d
-          else d.copy(body = newBody)
+          else d.copy(body = newBody)(d.isTailRec)
```

Comment:
To avoid Boolean confusion in case someone adds another parameter later:
```suggestion
          else d.copy(body = newBody)(isTailRec = d.isTailRec)
```